python/lsst/pipe/base/graphBuilder.py

```diff
@@ -279,57 +314,37 @@ def __repr__(self):
     """The dimensions of a single `Quantum` of this task (`DimensionGraph`).
     """
 
-    initInputs: _DatasetScaffoldingDict
+    initInputs: _DatasetDict
     """Dictionary containing information about datasets used to construct this
-    task (`_DatasetScaffoldingDict`).
+    task (`_DatasetDict`).
     """
 
-    initOutputs: _DatasetScaffoldingDict
+    initOutputs: _DatasetDict
     """Dictionary containing information about datasets produced as a
-    side-effect of constructing this task (`_DatasetScaffoldingDict`).
+    side-effect of constructing this task (`_DatasetDict`).
     """
 
-    inputs: _DatasetScaffoldingDict
+    inputs: _DatasetDict
     """Dictionary containing information about datasets used as regular,
-    graph-constraining inputs to this task (`_DatasetScaffoldingDict`).
+    graph-constraining inputs to this task (`_DatasetDict`).
     """
 
-    outputs: _DatasetScaffoldingDict
+    outputs: _DatasetDict
     """Dictionary containing information about datasets produced by this task
-    (`_DatasetScaffoldingDict`).
+    (`_DatasetDict`).
     """
 
-    prerequisites: _DatasetScaffoldingDict
+    prerequisites: _DatasetDict
     """Dictionary containing information about input datasets that must be
     present in the repository before any Pipeline containing this task is run
-    (`_DatasetScaffoldingDict`).
-    """
-
-    dataIds: Set[ExpandedDataCoordinate]
-    """Data IDs for all quanta for this task in the graph (`set` of
-    `ExpandedDataCoordinate`).
-
-    Populated after construction by `_PipelineScaffolding.fillDataIds`.
+    (`_DatasetDict`).
     """
 
-    quanta: List[Quantum]
-    """All quanta for this task in the graph (`list` of `Quantum`).
-
-    Populated after construction by `_PipelineScaffolding.fillQuanta`.
+    quanta: Dict[DataCoordinate, _QuantumScaffolding]
```

Comment:
Are you bothered by the fact that this is a mapping and not some other kind of collection, or the fact that the value type is not `Quantum`?  I'm more sympathetic to the latter, but it's the pattern I've followed elsewhere in this file (e.g. "tasks" is a collection of `_TaskScaffolding`), and I think alternative is a lot of "Scaffolding" suffixes on variables.  That feels like visual noise to me, because I'm already familiar with this code, but it probably wouldn't be to someone reading it for the first time, and I guess the question is who to optimize readability for.  This code isn't going to be understandable at a glance regardless, so I lean towards making it more readable for those who have already put some time into studying it.

Reply:
the latter, as it is just hard to understand what to expect from reading variable names places. It is probably not worth a large overhaul around, but something to keep in mind.